bigbio/utils/license.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Define LICENCE classes to normalize _LICENSE tag of dataloader
+"""
+
+from dataclasses import dataclass
+from typing import Optional
+
+
+class LicenseMixin:
+    """
+    Base class from which all licenses inherit
+    """
+
+    def to_string(self) -> str:
+        """
+        Get formatted name
+        """
+        text = f"{self.name}"
+
+        if hasattr(self, "type") and getattr(self, "type", None) is not None:
+            text += f" - {self.type}"
+
+        if hasattr(self, "version") and getattr(self, "version", None) is not None:
+            text += f" - v{self.version}"
+
+        if hasattr(self, "link") and getattr(self, "link", None) is not None:
+            text += f" ({self.link})"
+
+        if (
+            hasattr(self, "description")
+            and getattr(self, "description", None) is not None
+        ):
+            text += f": {self.description}"
+
+        return text
+
+    def __repr__(self):
+        """
+        Override to get a formatted full description of the license
+        """
+        return self.to_string()
+
+    def __str__(self):
+        """
+        Override to get a formatted full description of the license
+        """
+        return self.to_string()
+
+
+@dataclass
+class CreativeCommons(LicenseMixin):
```

Comment:
do you imagine "mixing in" more than one class here? just an idea, but could we get away with a single `LICENSE` dataclass here and create instances of it that we can use elsewhere? 
e.g. 
```
CREATIVE_COMMONS_BY_NC_SA_4p0 = LICENSE(
  name = "CC BY-NC-SA" # also maybe chosen from an enum
  family = "CC" # maybe chosen from an enum
  version = "4.0"
  )
```